src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final double side;
+    private final ColorSupplier color;
+
+    public Square(ColorSupplier color, double side) {
+        this.color = color;
+        this.side = side;
+    }
+
+    @Override
+    public double calculateArea() {
+        return side * side;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: square, area: "
+                + calculateArea() + "sq. units, side: "
+                + side + " units, "
+                + "color: " + color);
```

Comment:
Use only eng in messages/code: Instead of using the 'ColorSupplier' object directly, you should use a method to get the color name as a string.